src/app/actions.ts

```diff
@@ -130,4 +132,75 @@ export const getScoreFactors = async () => {
   const data = await res.json();
 
   return ScoreFactorsSchema.parse(data);
+};
+
+export const getYoutubeChannelUploadsPlaylistId = async (searchParams: { channelId?: string; channelUsername?: string }) => {
+  const { channelId, channelUsername } = searchParams || {};
+
+  // Validate inputs: at least one of channelId or channelUsername is required
+  if (!channelId && !channelUsername) {
+    throw new Error('Validation error: either channelId or channelUsername must be provided.');
+  }
+
+  const url = new URL('https://www.googleapis.com/youtube/v3/channels');
+
+  url.searchParams.set('part', 'contentDetails');
+
+  if (channelId) url.searchParams.set('id', channelId);
+  if (channelUsername) url.searchParams.set('forUsername', channelUsername);
+
+  url.searchParams.set('key', ENV.YOUTUBE_API_KEY);
+
+  const res = await fetch(url.toString());
+
+  // Check HTTP response
+  if (!res.ok) {
+    const bodyText = await res.text();
+    throw new Error(
+      `YouTube API error: ${res.status} ${res.statusText}${bodyText ? ` - ${bodyText}` : ''}`
+    );
+  }
+
+  const data = await res.json();
+
+  const uploads = data.items[0]?.contentDetails?.relatedPlaylists?.uploads;
+  if (!uploads) {
+    throw new Error('Channel found but uploads playlist ID is missing in response.');
+  }
+
+  return YoutubePlaylistIdSchema.parse(uploads);
+};
+
+export const getYoutubePlaylistVideos = async (playlistId: string, maxResults: number = 50) => {
+  // Clamp maxResults to YouTube API allowed range [1..50]
```

Comment:
LGTM the only thing I see is that this line of code seems too defensive like if the function was going to be used by a malicious coder haha so maybe being to complex to avoid breaking it if called with NaN for example ?